scripts/install-macos.sh

```diff
@@ -0,0 +1,268 @@
+#!/bin/bash
+# shellcheck disable=SC2016
+set -e
+
+# Set this machine's dependencies status(es) with the OK flag
```

Comment:
can you add to this comment that this variable is what we use as the return value from functions. we do so by updating the value of the OK variable before we return from the function and the calling code can read the OK variable.